client/src/templates/Challenges/ms-trophy/show.tsx

```diff
@@ -82,16 +82,34 @@ interface MsTrophyProps {
   updateChallengeMeta: (arg0: ChallengeMeta) => void;
 }
 
-// Component
-class MsTrophy extends Component<MsTrophyProps> {
-  static displayName: string;
-  private container: React.RefObject<HTMLElement> = React.createRef();
-
-  constructor(props: MsTrophyProps) {
-    super(props);
-    this.handleSubmit = this.handleSubmit.bind(this);
-  }
-  componentDidMount() {
+function MsTrophy(props: MsTrophyProps) {
+  const container = useRef<HTMLElement>(null);
+  const {
+    challengeMounted,
+    data: {
+      challengeNode: {
+        challenge: {
+          fields: { tests },
+          title,
+          challengeType,
+          helpCategory
+        }
+      }
+    },
+    pageContext: { challengeMeta },
+    initTests,
+    updateChallengeMeta
+  } = props;
+  initTests(tests);
+  updateChallengeMeta({
+    ...challengeMeta,
+    title,
+    challengeType,
+    helpCategory
+  });
+  challengeMounted(challengeMeta.id);
+  container.current?.focus();
```

Comment:
```suggestion
```
There's no need to call these on every render. This should map to the original `componentDidMount` callback, i.e. get called once when the component is first created.